frontend/src/pages/PipelineDetails.tsx

```diff
@@ -199,51 +231,67 @@ class PipelineDetails extends Page<{}, PipelineDetailsState> {
     let selectedVersion: ApiPipelineVersion | undefined;
     let versions: ApiPipelineVersion[] = [];
 
-    // If fromRunId is specified, load the run and get the pipeline template from it
-    if (fromRunId) {
+    // If fromRunId or fromRecurringRunId is specified, load the run and get the pipeline template from it
+    if (origin) {
+      const msgRunOrRecurringRun = origin.isRecurring ? 'recurring ' : '';
```

Comment:
nit: `? 'recurring run' : 'run'`
make the msg self-contained, so you can have `${msgRunOrRecurringRun}` instead of `${msgRunOrRecurringRun}run` below.